src/awkward/_broadcasting.py

```diff
@@ -685,9 +685,9 @@ def continuation():
                         if isinstance(x, RegularArray):
                             if maxsize > 1 and x.size == 1:
                                 nextinputs.append(
-                                    IndexedArray(
-                                        tmpindex, x.content[: x.length * x.size]
-                                    ).project()
+                                    x.content[: x.length * x.size]._carry(
```

Comment:
I can't see the arguments of this `_carry` above (on my phone), but if the second argument is `allow_lazy=True`, then a RecordArray will take that opportunity to put itself inside an IndexedArray, rather than actually carrying. When applying the optimization, we didn't replace all these `_carry` calls, we added the option. There are a few places where `allow_lazy=False` because without it, there would be an infinite loop. So we allow lazy as much as possible—this might be replacing code that always carried.